src/one-time-token/controller.ts

```diff
@@ -0,0 +1,39 @@
+import Controller from '@curveball/controller';
+import { Context } from '@curveball/core';
+import * as privilegeService from '../privilege/service';
+import { Forbidden, NotFound } from '@curveball/http-errors';
+import * as userService from '../user/service';
+import { User } from '../user/types';
+import { createToken } from '../reset-password/service';
+import * as hal from './formats/hal';
+
+class OneTimeTokenController extends Controller {
+
+  async post(ctx: Context<any>) {
+
+    if (!await privilegeService.hasPrivilege(ctx, 'admin')) {
+      throw new Forbidden('Only users with the "admin" privilege can request for one time token');
+    }
+
+    let user: User;
+    try {
+      user = await userService.findByIdentity(ctx.state.user.identity);
+    } catch (err) {
+      if (err instanceof NotFound) {
+        ctx.status = 404;
+        return;
```

Comment:
We don't set the status for errors, we throw exceptions. In this case you don't need to because the `NotFound` exception was already thrown. So all you have to do is *not* catch it.